src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -968,6 +968,13 @@ final class LazyList[+A] private(private[this] var lazyState: () => LazyList.Sta
   */
 @SerialVersionUID(3L)
 object LazyList extends SeqFactory[LazyList] {
+  // Override SeqFactory methods.
+  override def range[A: Integral](start: A, end: A): LazyList[A] =
+    range(start, end, step = Integral[A].one)
+
+  override def range[A](start: A, end: A, step: A)(implicit ev: Integral[A]): LazyList[A] =
+    LazyList.iterate(start)(ev.plus(_, step)).takeWhile(ev.lt(_, end))
```

Comment:
Is there a better way to implement this?
Should we add a check to ensure `start <= end`?

Reply:
> Should we add a check to ensure `start <= end`?

I don't think so, that's already handled by the `takeWhile` (also would be `start < end`)

> Is there a better way to implement this?

The following implementation allows `Integral[A].one` to be computed lazily for the `range` method that doesn't have a `step` parameter.

```suggestion
    newLL(rangeImpl(start, end, step))

  private[this] def rangeImpl[A](start: A, end: A, step: A)(implicit ev: Integral[A]): State[A] =
    if (ev.lt(start, end)) sCons(start, newLL(rangeImpl(ev.plus(start, step), end, step)))
    else State.Empty
```

then, the `range` method that doesn't have a `step` parameter should have the following implementation:

```scala
    newLL(rangeImpl(start, end, Integral[A].one))
```

also, someone yell at author of the laziness tests to make the one for `range` better